### DISCUSSION TITLE
api: library usage

### DISCUSSION THREAD

**Discussion by Wondertan** [celestiaorg/celestia-node#3220] [Ideas]
The node was designed with in-process programmatic API access in mind. However, the setup for that is currently complicated, and we don't even though if that works. So the goal of this issue is to understand what's left to make it work and clean up things up with optimization for ergonomics. We should also provide examples of how to achieve that.

**Comment by renaynay** [comment]
@distractedm1nd

**Comment by ramin** [comment]
example: https://github.com/cmwaters/blobusign/blob/main/node/node.go
